Add RN formatting information (#538)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/style_guidelines/release-notes.adoc

“Some referenced tickets are not linked. This means that the ticket is not accessible without Red Hat credentials."

[[release-note-asciidoc-format]]
== Release note formatting in AsciiDoc

> To display individual release notes outside of the structure of the table of contents, treat each release note as a description list item. This format is also compatible with the DITA/AEM migration. For additional formatting examples, you can inspect the link:https://redhat-documentation.github.io/supplementary-style-guide/#release-notes[Release notes] file in the `supplementary-style-guide` repository.

Reviewer: @jafiala I would rewrite the first sentence a bit, because I think that the reason for using description lists is not to have excessive headings, which are not allowed in DITA. 

The second sentence sends people to a link within the style guide, but the way it is currently worded, I thought I was going to a different location. How about writing this paragraph like this instead?

```suggestion
To avoid nesting headings excessively, treat each release note as a description list item. This format is compatible with the DITA/AEM migration. For additional formatting examples, see link:https://redhat-documentation.github.io/supplementary-style-guide/#release-notes[Release notes] entry in the `supplementary-style-guide`.
```

I know that we were discussing having a separate file or appendix with an ASCIIDoc example of a release notes file, but this doesn't seem to be what you're linking to. Am I missing something?

Author: ```suggestion
To avoid nesting headings excessively, treat each release note as a description list item. This format is also compatible with the DITA/AEM migration. For additional formatting examples, you can inspect the link:https://github.com/redhat-documentation/supplementary-style-guide/blob/main/supplementary_style_guide/style_guidelines/release-notes.adoc?plain=1[Release notes] file in the `supplementary-style-guide` repository.
```

> To avoid nesting headings excessively, treat each release note as a description list item. This format is also compatible with the DITA/AEM migration. For additional formatting examples, you can inspect the link:https://github.com/redhat-documentation/supplementary-style-guide/blob/main/supplementary_style_guide/style_guidelines/release-notes.adoc?plain=1[Release notes] file in the `supplementary-style-guide` repository.

Reviewer 2: ```suggestion
To avoid nesting headings excessively, treat each release note as a description list item. This format is also compatible with the DITA/AEM migration. For additional formatting examples, see the link:https://github.com/redhat-documentation/supplementary-style-guide/blob/main/supplementary_style_guide/style_guidelines/release-notes.adoc?plain=1[Release notes] file in the `supplementary-style-guide` repository.
```

.Release note AsciiDoc basic formatting template
----
Release note heading::
This is the main release note text.
+
Add another paragraph if necessary.
+
link:https://issues.redhat.com/browse/TICKET-REFERENCE[TICKET-REFERENCE]
----

> For the DITA conversion to work correctly, the list must remain uninterrupted. Attach all lines after the first to the description with a plus sign. Do not use open blocks (`--`) to separate elements within the description. If you need to introduce a list inside a release note text, add an empty line before attaching the reference line with a plus sign to preserve continuation.

Reviewer: Is "attach" the right word here? Maybe it should say something like this?

```suggestion
For the DITA conversion to work correctly, the list must remain uninterrupted. When creating the description list, put a plus sign before each item in the list. Do not use open blocks (`--`) to separate elements within the description. If you need to introduce a list inside a release note text, add an empty line before attaching the reference line with a plus sign to preserve continuation.
```
I'm also not sure what the "reference line" is. I think what you're saying is that you must have a blank line before you add more lines to the release note, but I'm not sure. I'm a little lost.

Reviewer: I think we'll need to discuss this on Monday.

> For the DITA conversion to work correctly, the list must remain uninterrupted. If the release note needs another paragraph or additional elements, such as a bullet list, you must attach all lines after the first line to the description with a plus sign. Do not use open blocks (`--`) to separate elements within the description. If you need to have a list inside a release note text, add an empty line followed by a plus sign after the list to attach the next paragraph, such as the ticket reference.

Author: @jhradilek , please check https://github.com/redhat-documentation/supplementary-style-guide/pull/538/commits/52046dd4579bb9e2b6b092849ca970807b15cb7d

> For the DITA conversion to work correctly, the list must remain uninterrupted. If the release note needs another paragraph or additional elements, such as a bullet list, you must attach all lines after the first line to the description with a plus sign. Do not use open blocks (`--`) to separate elements within the description. If you need to have a list inside a release note text, add an empty line followed by a plus sign after the list to attach the next paragraph, such as the ticket reference.

Reviewer 3: @jafiala I think that will work in most cases. The real point is to double check that the markup does not accidentally interrupt the list, but I understand why that is a more complicated instruction.

> For the DITA conversion to work correctly, the list must remain uninterrupted. If the release note needs another paragraph or additional elements, such as a bullet list, you must attach all lines after the first line to the description with a plus sign. Do not use open blocks (`--`) to separate elements within the description. If you need to have a list inside a release note text, add an empty line followed by a plus sign after the list to attach the next paragraph, such as the ticket reference.

Reviewer 2: ```suggestion
For the DITA conversion to work correctly, the list must remain uninterrupted. If the release note needs another paragraph or additional elements, such as a bulleted list, you must attach all lines after the first line to the description with a plus sign. Do not use open blocks (`--`) to separate elements within the description. If you need to have a list inside a release note text, add an empty line followed by a plus sign after the list to attach the next paragraph, such as the ticket reference.
```
It is very difficult to follow the instructions here. You also give different instructions for a bulleted list vs. a list, which is confusing because a bulleted list is a type of list. I think part of the issue is that you are giving too many rules in one paragraph. Give the rule for adding additional elements with a plus sign and an example, then the rule about adding a list and an example.

Author: How about this, is it clearer?

```suggestion
For the DITA conversion to work correctly, the list must remain uninterrupted. If the release note needs another paragraph or additional elements, you must attach all lines after the first one with a plus sign. If you need to have a list inside a release note text, attach it with a plus sign, and add an empty line followed by a plus sign after the list to attach the next paragraph, such as the ticket reference. Do not use open blocks (`--`) to separate elements within the description.
```

.Release note AsciiDoc complex formatting template
----

> Release note heading::

Author: This is something I keep forgetting, thanks.
